src/commands/crashlytics-mappingfile-generateid.ts

```diff
@@ -0,0 +1,45 @@
+import { Command } from "../command";
+import * as utils from "../utils";
+
+import { fetchBuildtoolsJar, runBuildtoolsCommand } from "../crashlytics/buildToolsJarHelper";
+import { Options } from "../options";
+import { FirebaseError } from "../error";
+
+interface CommandOptions extends Options {
+  resourceFile: string;
+}
+
+interface JarOptions {
+  resourceFilePath: string;
+}
+
+export const command = new Command("crashlytics:mappingfile:generateid")
+  .description(
+    "generate a mapping file id and write it to an Android resource file, which will be built into the app"
+  )
+  .option(
+    "--resource-file <resourceFile>",
+    "Path to the Android resource XML file that will be created or updated."
```

Comment:
general guidance around descriptions and options: https://github.com/firebase/firebase-tools/blob/master/CONTRIBUTING.md#create-a-file-for-your-command

```suggestion
    "path to the Android resource XML file that will be created or updated"
```